src/components/OrderDetailWidget/OrderDetailWidget.tsx

```diff
@@ -119,11 +119,13 @@ const OrderDetailWidget: FC<OrderDetailWidgetProps> = ({ order }) => {
 
   const [showFeeInfo, toggleShowFeeInfo] = useToggle(false);
 
+  const routerState = history.location.state as OrderDetailRouterState;
+  const shopwViewAllQuotes =
+    routerState?.fromSwapFlow === true && !userIsMakerOfSwap;
+
   // button handlers
   const handleBackButtonClick = () => {
-    history.push({
-      pathname: `/${userOrders.length ? AppRoutes.myOrders : AppRoutes.make}`,
-    });
+    history.goBack();
```

Comment:
```suggestion
    if (userIsFromSwapFlow) {
      history.goBack();

      return;
    }

    history.push({
      pathname: `/${userOrders.length ? AppRoutes.myOrders : AppRoutes.make}`,
    });
```

Reply:
I did this because else you will route away from the website when you land via a direct link.